src/osd/ECTransaction.h

```diff
@@ -120,6 +132,28 @@ class ECTransaction : public PGBackend::PGTransaction {
     assert(len == bl.length());
     ops.push_back(AppendOp(hoid, off, bl, fadvise_flags));
   }
+  void write(
+    const hobject_t &hoid,
+    uint64_t off,
+    uint64_t len,
+    bufferlist &bl,
+    uint32_t fadvise_flags) {
+    if (len == 0) {
+        touch(hoid);
+        return;
+    }
+    offset_write = true;
+
+    written += off + len > written ? (off + len - written) : 0;
+    assert(len == bl.length());
+    writeops.push_back(WriteOp(hoid, off, len, bl, fadvise_flags));
+  }
+  WriteOp* get_writeop() {
+    assert(!writeops.empty());
+    WriteOp* op = &(writeops.front());
```

Comment:
Sorry, missed a logic here, it should traverse the writeops and put it into backend's queue.